src/vivliostyle/table.js

```diff
@@ -1225,19 +1350,43 @@ goog.scope(function() {
     };
 
     /**
+     * @override
+     */
+    function adjustCellHeight(cellFragment, formattingContext) {
+        var repetitiveElements = formattingContext.getRepetitiveElements();
+        if (!repetitiveElements) return;
+
+        var vertical = formattingContext.vertical;
+        var column = cellFragment.pseudoColumn.column;
+        var cellContentElement = column.element;
+        var cellElement = cellFragment.cellNodeContext.viewNode;
+
+        var cellElementRect = column.clientLayout.getElementClientRect(cellElement);
+        if (vertical) {
+            var width = (cellElementRect.right - column.footnoteEdge - repetitiveElements.calculateElementHeight());
+            adapt.base.setCSSProperty(cellContentElement, "width", width + "px");
+        } else {
+            var height = (column.footnoteEdge - repetitiveElements.calculateElementHeight() - cellElementRect.top);
+            adapt.base.setCSSProperty(cellContentElement, "height", height + "px");
```

Comment:
There is another bug that `vertical-align` for a table cell is not effective when the cell is fragmented (see `table_vertical_align.html` for example).
Using `max-height`/`max-width` solves this bug too.